startup/11-optics.py

```diff
@@ -0,0 +1,173 @@
+print(f'Loading {__file__}...')
+
+import numpy as np
+from ophyd import EpicsMotor, EpicsSignal, EpicsSignalRO
+from ophyd import Device
+from ophyd import Component as Cpt
+from ophyd import PVPositionerPC
+from nslsii.devices import TwoButtonShutter
+
+
+### Setup photon shutters
+shut_fe = TwoButtonShutter('XF:05ID-PPS{Sh:WB}', name='shut_fe')
+shut_a = TwoButtonShutter('XF:05IDA-PPS:1{PSh:2}', name='shut_a')
+shut_b = TwoButtonShutter('XF:05IDB-PPS:1{PSh:4}', name='shut_b')
+
+
+### Setup white/pink beam slits
+class SRXSlitsWB(Device):
+    # Real synthetic axes
+    h_cen = Cpt(EpicsMotor, 'XCtr}Mtr')
+    h_gap = Cpt(EpicsMotor, 'XGap}Mtr')
+    v_cen = Cpt(EpicsMotor, 'YCtr}Mtr')
+    v_gap = Cpt(EpicsMotor, 'YGap}Mtr')
+
+    # Real motors
+    top = Cpt(EpicsMotor, 'T}Mtr')
+    bot = Cpt(EpicsMotor, 'B}Mtr')
+    inb = Cpt(EpicsMotor, 'I}Mtr')
+    out = Cpt(EpicsMotor, 'O}Mtr')
+
+class SRXSlitsPB(Device):
+    # Real synthetic axes
+    h_cen = Cpt(EpicsMotor, 'XCtr}Mtr')
+    h_gap = Cpt(EpicsMotor, 'XGap}Mtr')
+
+    # Real motors
+    inb = Cpt(EpicsMotor, 'I}Mtr')
+    out = Cpt(EpicsMotor, 'O}Mtr')
+
+slt_wb = SRXSlitsWB('XF:05IDA-OP:1{Slt:1-Ax:', name='slt_wb')
+slt_pb = SRXSlitsPB('XF:05IDA-OP:1{Slt:2-Ax:', name='slt_pb')
+
+# Is this used? Commented out before removing
+# class SRXSlits2(Device):
+#     inb = Cpt(EpicsMotor, 'I}Mtr')
+#     out = Cpt(EpicsMotor, 'O}Mtr')
+
+
+### Setup HFM Mirror
+class SRXHFM(Device):
+    x = Cpt(EpicsMotor, 'X}Mtr')
+    y = Cpt(EpicsMotor, 'Y}Mtr')
+    pitch = Cpt(EpicsMotor, 'P}Mtr')
+    bend = Cpt(EpicsMotor, 'Bend}Mtr')
+
+hfm = SRXHFM('XF:05IDA-OP:1{Mir:1-Ax:', name='hfm')
+
+
+### Setup HDCM
+class HDCMPIEZOROLL(PVPositionerPC):
+    setpoint = Cpt(EpicsSignal, '')
+    readback = Cpt(EpicsSignalRO, '')
+    pid_enabled = EpicsSignal('XF:05IDD-CT{FbPid:01}PID:on')
```

Comment:
Doing this in bad practice, this will instantiate the Signal instance on the class (not on demand on the instance) and the signal won't participate in the Device component logic.

https://github.com/NSLS-II-SMI/profile_collection/blob/a3f008b8d2f2715f14b31ce1a4d458793c22c45e/startup/09-machine.py#L68-L70  is an example on how to fix this.

`..., add_prefix=(),...`